src/moleculesampleapp/Arrow.java

```diff
@@ -0,0 +1,112 @@
+package moleculesampleapp;
+
+import javafx.geometry.Point3D;
+import javafx.scene.Group;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.PhongMaterial;
+import javafx.scene.shape.*;
+import javafx.scene.transform.Rotate;
+
+/**
+ * Created by Piotr on 23/01/2016.
+ */
+public class Arrow extends Group {
+    private int len, radius, tipRadius, tipLen;
+
+    private int divisions;
+    private TriangleMesh tipMesh;
+    private MeshView tipView;
+    private PhongMaterial arrowMaterial;
+
+    private Cylinder line;
+
+    Arrow() {
+        this(300, 50, 100, 100);
+    }
+
+    Arrow(int len_, int radius_, int tipRadius_, int tipLen_) {
+        super();
+        this.len = len_;
+        this.radius = radius_;
+        this.tipRadius = tipRadius_;
+        this.tipLen = tipLen_;
+        this.divisions = 10;
+        this.arrowMaterial = new PhongMaterial();
+        this.arrowMaterial.setDiffuseColor(Color.RED);
+        this.arrowMaterial.setSpecularColor(Color.DARKRED);
+        this.generate();
+    }
+
+    public void setLen(int len_) {
+        this.len = len_;
+        this.generate();
+    }
+
+    public void setRadius(int radius_) {
+        this.radius = radius_;
+        this.generate();
+    }
+
+    public void setTipRadius(int tipRadius_) {
+        this.tipRadius = tipRadius_;
+        this.generate();
+    }
+
+    public void setTipLen(int tipLen_) {
+        this.tipLen = tipLen_;
+        this.generate();
+    }
+
+    public void setMaterial(PhongMaterial material) {
+        this.arrowMaterial = material;
+        this.generate();
+    }
+
+    public void setDivisions(int divisions_) {
+        this.divisions = divisions_;
+        this.generate();
+    }
+
+    private void generate() {
+        //Generate Arrow Tip (Cone)
+        this.tipMesh = new TriangleMesh();
+        this.tipView = new MeshView();
+        this.tipMesh.getTexCoords().addAll(0, 0);
+        this.tipMesh.getPoints().addAll(0, 0, tipLen);//top center
+
+        for (int division = 0; division < this.divisions; division++) {
+            double angle = Math.PI / (float) divisions * (float) division * 2.0;
+            this.tipMesh.getPoints().addAll(
+                    (int) (Math.sin(angle) * (double) tipRadius),
```

Comment:
:+1: the code in sub-class has also int cast so move to float would solve problems also there